sdk/include/opentelemetry/sdk/logs/exporter.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * ExportResult is returned as result of exporting a batch of Log Records.
+ */
+enum class ExportResult
+{
+  // The batch was exported successfully
+  kSuccess = 0,
+  // The batch was exported unsuccessfully and was dropped, but can not be retried
+  kFailure
+};
+
+/**
+ * LogExporter defines the interface that log exporters must implement.
+ */
+class LogExporter
+{
+public:
+  virtual ~LogExporter() = default;
+
+  /**
+   * Exports the batch of log records to their export destination.
+   * This method must not be called concurrently for the same exporter instance.
+   * The exporter may attempt to retry sending the batch, but should drop
+   * and return kFailure after a certain timeout.
+   * @param records a vector of unique pointers to log records
+   * @returns an ExportResult code (whether export was success or failure)
+   */
+  virtual ExportResult Export(
+      const std::vector<std::unique_ptr<opentelemetry::logs::LogRecord>> &records) noexcept = 0;
+
+  /**
+   * Marks the exporter as ShutDown and cleans up any resources as required.
+   * Shutdown should be called only once for each Exporter instance.
+   * @param timeout minimum amount of microseconds to wait for shutdown before giving up and
+   * returning failure.
+   * @return true if the exporter shutdown succeeded, false otherwise
+   */
+  virtual bool Shutdown(
+      std::chrono::microseconds timeout = std::chrono::microseconds::max()) noexcept = 0;
```

Comment:
In the tracing SDK, we use `0` as default timeout, here we use `std::chrono::microseconds::max()`. What is your reasoning behind that?

Reply:
I wanted to set a defined timeout (and chose the max possible value for it) instead of allowing the shutdown to hang forever, as the spec prohibits `Forceflush` and `Shutdown` from blocking indefinitely

Reply:
But open to discussion on this, also related to this [comment](https://github.com/open-telemetry/opentelemetry-cpp/pull/403#discussion_r529887038)

Reply:
Then let's submit an issue so we can make this consistent in the different exporter interfaces. It's not intuitive to have `0` as default in the trace exporter and `max()` as default in the log exporter. We should settle on one default across exporters.

Reply:
Flagged in https://github.com/open-telemetry/opentelemetry-cpp/issues/418